opentelemetry-sdk/src/trace/span_processor.rs

```diff
@@ -712,16 +713,18 @@ mod tests {
             instrumentation_lib: Default::default(),
         };
         processor.on_end(unsampled);
-        assert!(!exporter.is_export_called());
+        assert!(exporter.get_finished_spans().unwrap().is_empty());
     }
 
     #[test]
     fn simple_span_processor_shutdown_calls_shutdown() {
-        let exporter = TestSpanExporter::new();
+        let exporter = InMemorySpanExporterBuilder::new().build();
         let mut processor = SimpleSpanProcessor::new(Box::new(exporter.clone()));
-        assert!(!exporter.is_shutdown_called());
+        let span_data = new_test_export_span_data();
+        processor.on_end(span_data.clone());
+        assert_eq!(exporter.get_finished_spans().unwrap()[0], span_data);
         let _result = processor.shutdown();
-        assert!(exporter.is_shutdown_called());
+        assert!(exporter.get_finished_spans().unwrap().is_empty());
```

Comment:
nit: Add a comment saying, we indirectly test shutdown is called, by asserting that spans are empty in inmemoryexporter.